src/makeInfinityAdder.js

```diff
@@ -4,7 +4,46 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let result = 0;
+
+  return function adder(a) {
+    if (a === undefined) {
+      result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += a;
+
+      result = sum;
+
+      return (b) => {
+        if (b === undefined) {
+          sum = 0;
+
+          return result;
+        } else {
+          sum += b;
+
+          result = sum;
+
+          return (c) => {
+            if (c === undefined) {
+              sum = 0;
+
+              return result;
+            } else {
+              sum += c;
+
+              result = sum;
+            }
+          };
+        }
+      };
+    }
+  };
```

Comment:
The current implementation uses nested functions to handle additional arguments, which is unnecessarily complex. Consider refactoring the code to use a single function that accumulates the sum and returns it when called without arguments. This will simplify the logic and make the code more readable.